src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -1629,237 +1624,223 @@ void GetVRFInputData(EnergyPlusData &state, bool &ErrorsFound)
+                                                 thisVrfSys.CoolPLFFPLR, // Curve index
+                                                 {1},                    // Valid dimensions
+                                                 RoutineName,            // Routine name
+                                                 cCurrentModuleObject,   // Object Type
+                                                 thisVrfSys.Name,        // Object Name
+                                                 cAlphaFieldNames(12));  // Field Name
             if (!ErrorsFound) {
                 //     Test PLF curve minimum and maximum. Cap if less than 0.7 or greater than 1.0.
-                MinCurveVal = 999.0;
-                MaxCurveVal = -999.0;
-                CurveInput = 0.0;
-                while (CurveInput <= 1.0) {
-                    CurveVal = CurveValue(state, state.dataHVACVarRefFlow->VRF(VRFNum).CoolPLFFPLR, CurveInput);
-                    if (CurveVal < MinCurveVal) {
-                        MinCurveVal = CurveVal;
-                        MinCurvePLR = CurveInput;
-                    }
-                    if (CurveVal > MaxCurveVal) {
-                        MaxCurveVal = CurveVal;
-                        MaxCurvePLR = CurveInput;
-                    }
-                    CurveInput += 0.01;
-                }
+                auto [MinCurvePLR, MinCurveVal, MaxCurvePLR, MaxCurveVal] = checkCurveMinMaxOutput(thisVrfSys.CoolPLFFPLR);
+
                 if (MinCurveVal < 0.7) {
-                    ShowWarningError(
-                        state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + state.dataHVACVarRefFlow->VRF(VRFNum).Name + "\", invalid");
+                    ShowWarningError(state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + thisVrfSys.Name + "\", invalid");
                     ShowContinueError(state, "..." + cAlphaFieldNames(12) + "=\"" + cAlphaArgs(12) + "\" has out of range values.");
                     ShowContinueError(state,
                                       format("...Curve minimum must be >= 0.7, curve min at PLR = {:.2T} is {:.3T}", MinCurvePLR, MinCurveVal));
                     ShowContinueError(state, "...Setting curve minimum to 0.7 and simulation continues.");
-                    SetCurveOutputMinMaxValues(state, state.dataHVACVarRefFlow->VRF(VRFNum).CoolPLFFPLR, ErrorsFound, 0.7, _);
+                    SetCurveOutputMinMaxValues(state, thisVrfSys.CoolPLFFPLR, ErrorsFound, 0.7, _);
                 }
 
                 if (MaxCurveVal > 1.0) {
-                    ShowWarningError(
-                        state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + state.dataHVACVarRefFlow->VRF(VRFNum).Name + "\", invalid");
+                    ShowWarningError(state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + thisVrfSys.Name + "\", invalid");
```

Comment:
#AllTheFomats

Reply:
I've started doing this, I'm like 1 hour in, and I haven't left the block that deals with the VRF Sys type (so I need to do FluidCtrl and FluidCtrl_HR which are equally as big). I agree with all your comments, but this is a bug fix, so I'm afraid I won't address them.

I feel like we should be able to (at least partly) automate this, either via scripting or via clang-tidy (similar to https://github.com/mikecrowe/clang-tidy-fmt)

Reply:
I get it. I actually want to establish a different pattern entirely for IDF errors, so maybe this format stuff and that will happen in one pass.

Reply:
cf 7dbcaa0db9cde14a3b4a86f9acac005f8a9b92c3 . It's not impossible that I have done a couple of misteps, I am way too tired to properly check that I've done all these format replacements correctly.